File: gestionApp/personas/Empleado.java
package gestionApp.personas;

import gestionApp.Empresa;
import gestionApp.Negocio;
import gestionApp.servicios.Servicio;
import gestionApp.Venta;

import java.io.Serializable;
import java.util.ArrayList;

public class Empleado  extends Persona implements Serializable {
    private int cedula;
    private String id;
    private String cargo;
    private  String fechaInicioContrato;
    private  double habilidadVenta;
    public Empresa empresa;
    private  ArrayList<Servicio> servicios;
    private  ArrayList<Negocio> negociosEmpleado;
    private  ArrayList<Venta> ventas;
	private static int idCounter = 0;
    private static ArrayList<Empleado> allEmpleados = new ArrayList<>();

    public Empleado(String nombre, int cedula, String cel, String correo , String cargo, String fechaInicioContrato){
        super(nombre, cedula, cel, correo);
        this.id = "0".repeat(3-String.valueOf(idCounter).length()) + String.valueOf(idCounter);
        this.cargo = cargo;
        this.fechaInicioContrato = fechaInicioContrato;
        this.servicios = new ArrayList<>();
        this.negociosEmpleado = new ArrayList<>();
        this.ventas = new ArrayList<>();
        this.habilidadVenta = Math.random(); //numero random entre 0 y 0.4
        idCounter += 1;
        allEmpleados.add(this);
        idCounter += 1;
    }

    //---------------Metodos Heredados---------------

	public String getId(){return this.id;}

    public String getNombre(){return nombre;}
    public void setNombre(String nombre){this.nombre = nombre;}

    public int getCedula(){return cedula;}
    public void setCedula(int cedula){this.cedula = cedula;}

    public String getCel(){return this.cel;}
    public void setCel(String cel){this.cel = cel;}

    public String getCorreo(){return correo;}
    public void setCorreo(String correo){this.correo = correo;}

    public double getHabilidadVenta() {
        return habilidadVenta;
    }

    //---------------Metodos Propios---------------


    public String getCargo(){return cargo;}
    public void setCargo(String cargo){this.cargo = cargo;}

    public String getFechaInicioContrato (){return this.fechaInicioContrato;}

    public ArrayList<Servicio> getServicios(){return servicios;}
    public void setServicios(Servicio nuevoServicio){this.servicios.add(nuevoServicio);}

    public ArrayList<Negocio> getNegociosEmpleado(){return negociosEmpleado;}
    public void setNegociosEmpleado(Negocio nuevoNegocio){this.negociosEmpleado.add(nuevoNegocio);}

    public ArrayList<Venta> getVentas(){return ventas;}
    public void setVentas(Venta nuevaVenta){this.ventas.add(nuevaVenta);}
    
    public static ArrayList<Empleado> getAllEmpleados(){return allEmpleados;}

    public Empresa getEmpresa(){return empresa;}
	public void setEmpresa(Empresa empresa){this.empresa = empresa;}

    public String toStringEmpresa(){
		return "\n////////////////////////////\n" +
		"Empresa: " + getEmpresa().getNombre() +
		"\nNit Empresa: " + getEmpresa().getNit() +
		"\nDescripcion de la empresa a la que pertenece el Empleado: \n" + 
		getEmpresa().getDescripcion() +
		"\n////////////////////////////\n";  
	}
	@Override
	
	public  String toString(){
		return "\tDatos del Empleado \nCliente: "+ nombre +
		 "\nIdentificado con el CC." + cedula + 
		 "\nCelular: " + cel +
		 "\nCorreo: " + correo +
		 "\nID: " + id +
		 "\nCargo: " + cargo +
		 "\nNegocios del Empleado" +negociosEmpleado +
		 "\n/////////////////////////////////////////////\n";
		 
		
	}
}

File: Python/gestionApp/Personas/Empleado.py
import math
from random import uniform
from tkinter import messagebox

from Python.gestionApp.personas.Persona import Persona


class Empleado(Persona):
    _allEmpleados = []
    _servicios = []
    _negociosEmpleado = []
    _ventas = []
    _idCounter = 0

    def __init__(self, nombre, cedula, cel, correo, cargo, fechaContrato):
        super().__init__(nombre, cedula, cel, correo)

        self.cargo = cargo
        self.fechaContrato = fechaContrato
        self.habilidadVenta = uniform(0, 0.4)

        Empleado._allEmpleados.append(self)
        messagebox.showinfo(message = 'EMPLEADO creada correctamente', title = 'Creacion Empleado')




    def getCargo(self): return self.cargo
    def setCargo(self, cargo): self.cargo = cargo

    def getFechaContrato(self): return self.fechaContrato
    def setFechaContrato(self, fechaContrato): self.fechaContrato = fechaContrato

    def getHabilidadVenta(self): return self.habilidadVenta

    @classmethod
    def getAllEmpleados(cls): return cls._allEmpleados

    @classmethod
    def setAllEmpleados(cls, allEmpleados): cls._allEmpleados = allEmpleados

    @classmethod
    def getServicios(cls): return cls._servicios

    @classmethod
    def setServicios(cls, servicios): cls._servicios = servicios

    @classmethod
    def getNegociosEmpleado(cls): return cls._negociosEmpleado

    @classmethod
    def setNegociosEmpleado(cls, negocio): cls._negociosEmpleado.append(negocio)
    
    def busquedaEmpleado(cedula):
        for i in range(len(Empleado.getAllEmpleados())):
            if (Empleado.getAllEmpleados(Empleado.getCedula(i))):
                return Empleado.getAllEmpleados(i)      
            
    def infoEmpleado(self):
        return f'\t---Informacion Empleado---  \n Nombre: {self.nombre} \n Cedula: {self.cedula} \n Celular: {self.celular} \n Cargo: {self.cargo} \n Fecha Contrato: {self.fechaContrato}'

               
    
    